arcaflow_plugin_rtla/rtla_plugin.py

```
@@ -92,9 +119,14 @@ def run_timerlat(
         # In either the case of a keyboard interrupt or a cancel signal, we need to
         # send the SIGINT to the subprocess.
         if self.finished_early:
-            proc.send_signal(2)
+            timerlat_proc.send_signal(2)
 
-        output, _ = proc.communicate()
+        if params.enable_time_series and trace_file_exists:
+            # Interrupt the time series collection process and capture the output
```

I would base this test on whether the subprocess, `timeseries_proc` exists -- _that's_ what we want to signal, not the file, and that's the value that we're going to use to send the signal.

---

Not a bad idea. `timerlat_proc.send_signal(0)` should raise an `OSError` if the pid is invalid or doesn't exist... (although this will somewhat depend on exactly how `timerlat_proc.pid` is managed under the covers, and I haven't experimented).

---

I was actually thinking more about the memory object, `timeseries_proc`, as opposed to the process which it represents (especially since we would need a valid object in order to invoke `send_signal(0)`, and, if we have one, it's reasonable to assume that the process was successfully created).

In any case, sending "signal zero" seems like a good idea except for the fact that the next thing we intend to do is to send signal `2` -- we might as well skip the preamble once we've decided to try sending a signal.